delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/indexing/ElasticSearchIndexingCoordinator.scala

```diff
@@ -0,0 +1,157 @@
+      stream = eLog
+                 .evalMapFilterValue {
+                   case res if containsSchema(res) && containsTypes(res) => deleteOrIndex(res).map(Some.apply)
+                   case res if containsSchema(res)                       => delete(res).map(Some.apply)
+                   case _                                                => Task.pure(None)
+                 }
+                 .runAsyncUnit(bulk => IO.when(bulk.nonEmpty)(client.bulk(bulk).hideErrorsWith(illegalArgument)))
+                 .flatMap(Stream.chunk)
+                 .map(_.void)
+                 .persistProgress(initialProgress, projection, config.indexing.persist)
+    } yield stream
+}
+
+object ElasticSearchIndexingCoordinator {
+
+  private val logger: Logger = Logger[ElasticSearchIndexingCoordinator.type]
+
+  /**
+    * Create a coordinator for indexing documents into ElasticSearch indices triggered and customized by the ElasticSearchViews.
+    */
+  def apply(
+      views: ElasticSearchViews,
+      eventLog: GlobalEventLog[Message[ResourceF[IndexingData]]],
+      client: ElasticSearchClient,
+      projection: Projection[Unit],
+      config: ElasticSearchViewsConfig
+  )(implicit
+      as: ActorSystem[Nothing],
+      scheduler: Scheduler,
+      cr: RemoteContextResolution,
+      base: BaseUri
+  ): Task[IndexingStreamCoordinator[IndexingViewResource]] = {
+
+    val retryStrategy =
+      RetryStrategy[Throwable](config.indexing.retry, _ => true, logError(logger, "elasticsearch indexing"))
+
+    for {
+      coordinator <- IndexingStreamCoordinator[ResourceF[IndexingElasticSearchView]](
+                       "ElasticSearchViewsCoordinator",
+                       (res, progress) => new IndexingStream(client, res, config).build(eventLog, projection, progress),
+                       projection,
+                       config.processor,
+                       retryStrategy
+                     )
+      _           <- startIndexing(views, coordinator, config)
+    } yield coordinator
+  }
+
+  private def startIndexing(
+      views: ElasticSearchViews,
+      coordinator: IndexingStreamCoordinator[IndexingViewResource],
+      config: ElasticSearchViewsConfig
+  )(implicit as: ActorSystem[Nothing], sc: Scheduler) =
+    StreamSupervisor(
+      "ElasticSearchViewsIndexer",
+      streamTask = Task.delay(
+        views
+          .events(NoOffset)
+          .evalMapFilter { ev =>
+            views.fetch(IriSegment(ev.event.id), ev.event.project).map(Some(_)).redeemCause(_ => None, identity)
+          }
+          .evalMap {
+            case res @ ResourceF(_, _, _, _, false, _, _, _, _, _, view: IndexingElasticSearchView) =>
+              coordinator.start(res.as(view))
+            case res @ ResourceF(_, _, _, _, true, _, _, _, _, _, view: IndexingElasticSearchView)  =>
+              coordinator.stop(res.as(view))
```

Comment:
does this also delete the index?

Reply:
No it doesn't

Reply:
I have addressed the cases when we actually want to delete previous indices